test/specgen/swagger-spec-generator.test.js

```diff
@@ -517,6 +517,207 @@ describe('swagger definition', function() {
+      var swaggerResource = createSwaggerObject(app, {
+        generateRelationProperties: true,
+      });
+
+      // Model with relations
+      verifyResponseWithRelations(swaggerResource, 'get', [
+        '/Users'], 'UserWithRelations', true);
+
+      verifyResponseWithRelations(swaggerResource, 'get', [
+        '/Users/{id}',
+        '/Users/findOne',
+        '/Conversations/{id}/user',
+        '/Users/{id}/conversations/{nk}/user',
+      ], 'UserWithRelations', false);
+
+      verifyResponseWithRelations(swaggerResource, 'get', [
+        '/Users/{id}/conversations',
+        '/Conversations',
+      ], 'ConversationWithRelations', true);
+
+      verifyResponseWithRelations(swaggerResource, 'get', [
+        '/Users/{id}/conversations/{fk}',
+        '/Conversations/{id}',
+        '/Conversations/findOne',
+        '/Messages/{id}/conversation',
+      ], 'ConversationWithRelations', false);
+
+      verifyResponseWithRelations(swaggerResource, 'get', [
+        '/Conversations/{id}/messages',
+        '/Messages',
+        '/Users/{id}/conversations/{nk}/messages',
+      ], 'MessageWithRelations', true);
+
+      verifyResponseWithRelations(swaggerResource, 'get', [
+        '/Messages/{id}',
+        '/Conversations/{id}/messages/{fk}',
+        '/Users/{id}/conversations/{nk}/messages/{fk}',
+      ], 'MessageWithRelations', false);
+
+      // Model without relations
+      verifyResponseWithRelations(swaggerResource, 'post', [
+        '/Users',
+        '/Users/replaceOrCreate',
+        '/Users/upsertWithWhere',
+        '/Users/{id}/replace',
+      ], 'User', false);
+
+      verifyResponseWithRelations(swaggerResource, 'put', ['/Users'], 'User', false);
+
+      verifyResponseWithRelations(swaggerResource, 'patch', ['/Users'], 'User', false);
+
+      verifyResponseWithRelations(swaggerResource, 'post', [
+        '/Conversations',
+        '/Conversations/replaceOrCreate',
+        '/Conversations/upsertWithWhere',
+        '/Conversations/{id}/replace',
+      ], 'Conversation', false);
+
+      verifyResponseWithRelations(swaggerResource, 'put', [
+        '/Users/{id}/conversations/{fk}',
+      ], 'Conversation', false);
+
+      verifyResponseWithRelations(swaggerResource, 'post', [
+        '/Users/{id}/conversations',
+      ], 'Conversation', false);
+
+      verifyResponseWithRelations(swaggerResource, 'post', [
+        '/Messages',
+        '/Messages/replaceOrCreate',
+        '/Messages/upsertWithWhere',
+        '/Messages/{id}/replace',
+        '/Users/{id}/conversations/{nk}/messages',
+      ], 'Message', false);
+
+      verifyResponseWithRelations(swaggerResource, 'post', [
+        '/Conversations/{id}/messages',
+      ], 'Message', false);
+
+      verifyResponseWithRelations(swaggerResource, 'put', [
+        '/Conversations/{id}/messages/{fk}',
+        '/Users/{id}/conversations/{nk}/messages/{fk}',
+      ], 'Message', false);
+    });
+
+    it('honor disableInclude options', function() {
```

Comment:
When I was skimming through the tests, I though `disableInclude` was an option for the swagger gen, similar to `generateRelationProperties`. I was surprised that these tests are not setting it anywhere. Then I realized it's a model setting. Let's improve the test title to make this clear.

```suggestion
    it('honors model relation setting "disableInclude"', function() {
```